### PR TITLE
Add category context activity

### PR DESCRIPTION
**Pull request by garemoko**
This PR is actually good as a stand alone without the other two. Addresses part of https://github.com/jlowe64/moodle-logstore_xapi/issues/25

Again, still needs docs and tests.

### CODE REVIEW

**Comment by garemoko** [comment]
@ryansmith94 @jlowe64 this is ready for review. I haven't done anything with the tests on this one. Let me know if you think I need to.

**Comment by ryansmith94** [comment]
I think it does need tests in [EventTest.php](https://github.com/LearningLocker/xAPI-Recipe-Emitter/blob/master/tests/EventTest.php). It's quite easy for a new contributor (or even us) to accidentally overwrite the parts of the base Event, so we should add tests for these changes to prevent accidentally overwriting parts of the category.

**Comment by garemoko** [comment]
Ok, I think this is good now. @ryansmith94 please check these tests give the coverage you were expecting.

File: tests/UserRegisteredTest.php [hunk 1]
```diff
@@ -15,7 +15,7 @@ public function setup() {
     protected function constructInput() {
         return array_merge(
             parent::constructInput(),
-            $this->contructObject('app')
+            $this->contructObject('app', 'http://id.tincanapi.com/activitytype/site')
```
Comment (ryansmith94) [inline tests/UserRegisteredTest.php]: I think we should create a constructApp method instead in the EventTest.

File: tests/EventTest.php [hunk 2]
```diff
@@ -172,4 +198,9 @@ protected function assertAttempt($input, $output) {
         $this->assertArrayHasKey($input['attempt_ext_key'], $output['definition']['extensions']);
         $this->assertEquals($input['attempt_ext'], $output['definition']['extensions'][$input['attempt_ext_key']]);
     }
+
+    protected function assertContextActivities($input, $output) {
+        $this->assertEquals($input['contextActivities_moodleCatgeory'], $output['category'][0]);
```
Comment (ryansmith94) [inline tests/EventTest.php]: I'm not sure I completely understand this. None of the keys passed from the translator to the emitter should be Moodle specific either, so they definitely shouldn't have the word "moodle" in them. I can't see where `contextActivities_moodleCatgeory` comes from the translator to here either.

Comment (ryansmith94) [inline tests/EventTest.php]: I don't think this method needs to take `input` at all. I think we should just assert that the first category (or perhaps the last) contains the correct information.

**Comment by ryansmith94** [comment]
Ready to be merged @garemoko?

**Comment by garemoko** [comment]
@ryansmith94 yes, i think so!

**Comment by ryansmith94** [comment]
Awesome! I'll merge when I get home in about 30 minutes :-)

**Comment by ryansmith94** [comment]
It's only just occurred to me that this needs to be changed. We can't have Moodle specific things in this repo. Use the translator to pass through the context for Moodle. I'm going to have to either revert this merge or wait for another pull request that removes the Moodle specific parts.

File: src/Events/Event.php [hunk 3]
```diff
@@ -43,8 +43,25 @@ public function read(array $opts) {
                 ],
                 'contextActivities' => [
                     'grouping' => [
-                        $this->readApp($opts),
+                        $this->readApp($opts)
                     ],
+                    'category' => [
```
Comment (ryansmith94) [inline src/Events/Event.php]: This needs to be done differently. I'm not even certain that this should be here given that we have `readApp`.

File: tests/EventTest.php [hunk 4]
```diff
@@ -172,4 +190,23 @@ protected function assertAttempt($input, $output) {
         $this->assertArrayHasKey($input['attempt_ext_key'], $output['definition']['extensions']);
         $this->assertEquals($input['attempt_ext'], $output['definition']['extensions'][$input['attempt_ext_key']]);
     }
+
+    protected function assertContextActivities($output) {
+        $this->assertEquals([
```
Comment (ryansmith94) [inline tests/EventTest.php]: This needs to be done differently.

**Comment by ryansmith94** [comment]
We will be using this repository with applications other than Moodle.

**Comment by garemoko** [comment]
@ryansmith94 if you log this as an issue with some notes on how to address this, I'll take a look, possibly tomorrow. 

I guess we'd pull in the category activity name, description and id from the translator? Or would be go right back to the expander?

**Comment by ryansmith94** [comment]
@garemoko I've made comments on the files that need to be changed. You need to add some code for this in the base event of the Translator. No need to go back to the Expander. I'm not entirely sure what to call it in the translator. I think we should use the name `app` in the keys and rename `app` to `installation` or something. I don't want to call it `site` in case we use this for mobile apps rather than websites.

The other potential solution is that we change the `app_type` to `moodle` rather than `site` and remove all of this superfluous info about Moodle.

**Comment by ryansmith94** [comment]
Or we could use `app` and `source`.

**Comment by garemoko** [comment]
+1 to app and source

**Comment by ryansmith94** [comment]
Awesome, let's go with that then. Since we're on the topic of the "scope" of this repo, I should also point out that the Expander should be specific to Moodle and contain no references to the xAPI as it may be used with other standards. @garemoko

**Pull request merged**